### PR TITLE
Add App Check support to desktop Functions

### PR DESCRIPTION
**Pull request by a-maurice**
Add logic to include the App Check token to Functions calls on the desktop implementation. Adds Functions to the integration testapp.
***

Local testing
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [x] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

File: app_check/integration_test/functions/firebase.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "functions": {
+    }
```
Comment (AlmostMatt) [inline app_check/integration_test/functions/firebase.json:3]: What is this file? Can it be deleted?

**Review by AlmostMatt — APPROVED** [review]
Looks good, just a question about one file.

File: app_check/integration_test/functions/firebase.json (hunk 1 above)
Comment (a-maurice) [reply app_check/integration_test/functions/firebase.json:3]: I assumed it wasn't needed, but when I deleted it, the deploy failed, complaining about this file missing.  From the functions samples, it has logic like https://github.com/firebase/functions-samples/blob/main/2nd-gen/callable-functions/firebase.json, but we didn't need that additional logic for this simple function, hence why it is sorta empty.

Comment (AlmostMatt) [reply app_check/integration_test/functions/firebase.json:3]: Alright, keep it.

**Pull request merged**